server/src/jobs/auto-discovery.cpp

```diff
@@ -0,0 +1,275 @@
+            }
+            else {
+                return fty::unexpected("No host information retrieved from DNS");
+            }
+        }
+        else {
+            return fty::unexpected("Error during read DNS");
+        }
+    }
+    else {
+        return fty::unexpected("Ip address empty");
+    }
+    return {};
+}
+
+void AutoDiscovery::scan(const commands::discoveryauto::In in)
+{
+    std::string discoveryConfigFile("/etc/fty-discovery/fty-discovery.cfg");  // TBD
+    zconfig_t* config = zconfig_load(discoveryConfigFile.c_str());
+    if (!config) {
+        logError("Failed to load config file {}", discoveryConfigFile);
+        config = zconfig_new("root", NULL);
+    }
+
+    // Init default links TBD: To put in a function ???
+    fty::asset::create::PowerLinks defaultValuesLinks;
+    const char *CFG_DISCOVERY_DEFAULT_VALUES_LINKS("/defaultValuesLinks");
+    auto section = zconfig_locate(config, CFG_DISCOVERY_DEFAULT_VALUES_LINKS);
+    if (section) {
+        for (auto link = zconfig_child(section); link; link = zconfig_next(link)) {
+            std::string iname(zconfig_get(link, "src", ""));
+            fty::asset::create::PowerLink powerLink;
+            // When link to no source, the file will have "0"
+            powerLink.source = iname;
+            powerLink.link_type = fty::convert<uint32_t>(std::stoi(zconfig_get(link, "type", "1")));
+            if (!(powerLink.source.value() == "0")) {
+                defaultValuesLinks.append(powerLink);
+            }
+        }
+    }
+    logDebug("defaultValuesLinks size={}", defaultValuesLinks.size());
+
+    // Init default parent TBD: To put in a function ???
+    bool deviceCentricView = false;
+    std::string defaultParent("0");
+    const char *CFG_DISCOVERY_DEFAULT_VALUES_AUX("/defaultValuesAux");
+    section = zconfig_locate(config, CFG_DISCOVERY_DEFAULT_VALUES_AUX);
+    if (section) {
+        for (auto item = zconfig_child(section); item; item = zconfig_next(item)) {
+            std::string configName(zconfig_name(item));
+            // the config API call returns the parent internal name, while fty-proto-t needs to carry the database ID
+            if (configName == "parent") {
+                defaultParent = zconfig_value(item);
+                if (defaultParent.empty()) {
+                   defaultParent == "0";
+                }
+                if (defaultParent != "0") {
+                    deviceCentricView = true;
+                }
+            }
+        }
+    }
+    logDebug("defaultParent={}", defaultParent);
+
+    // Get list of protocols
+    commands::protocols::In in_prot;
+    in_prot.address = in.address;
+    Protocols protocols;
+    auto list_protocols = protocols.getProtocols(in_prot);
+    if (!list_protocols) {
+        logError(list_protocols.error().c_str());
+        return;
+    }
+    std::ostringstream list_str;
+    std::for_each((*list_protocols).begin(), (*list_protocols).end(),
```

Comment:
oh... why this echo from past? :)

Reply:
No need to be log. Decrease level to debug :-)